src/helpers/counterBuilder.js

```diff
@@ -0,0 +1,231 @@
+    this.state = {
+      ...this.state,
+      ...this.normalizeDigitLimits(minDigits, maxDigits)
+    }
+  }
+
+  /**
+   * Gets minDigits and maxDigits from props specified by the user.
+   * @return {array}
+   */
+  getDigitLimitsFromProps () {
+    var minDigits = {}
+    var maxDigits = {}
+
+    if (typeof this.props.minDigits === 'number') {
+      for (let period of this.state.periods) minDigits[period] = this.props.minDigits
+    } else if (typeof this.props.minDigits === 'object') {
+      minDigits = this.props.minDigits
+    }
+    if (typeof this.props.maxDigits === 'number') {
+      for (let period of this.state.periods) maxDigits[period] = this.props.maxDigits
+    } else if (typeof this.props.maxDigits === 'object') {
+      maxDigits = this.props.maxDigits
+    }
+    return [minDigits, maxDigits]
+  }
+
+  /**
+   * Calculates minimum number of digits for current radix and given period.
+   * @param {string} period - period to calculate minDigits for
+   * @return {number}
+   */
+  getMinSegmentSize (period) {
+    const maxValue = this.getPeriodMaxValue(period)
+    return (maxValue).toString(this.props.radix).length
+  }
+
+  /**
+   * Normalizes minDigits and maxDigits and sets default values where user did not specify them.
+   * @param {number|object} minDigits
+   * @param {number|object} maxDigits
+   * @return {object}
+   */
+  normalizeDigitLimits (minDigits, maxDigits) {
+    var normalizedMinDigits = {}
+    var normalizedMaxDigits = {}
+
+    for (let period of this.state.periods) {
+      const minSegmentSize = this.getMinSegmentSize(period);
+      [normalizedMinDigits[period], normalizedMaxDigits[period]] = this.normalizeDigitLimitsForPeriod(
+        minDigits[period], maxDigits[period], minSegmentSize
+      )
+    }
+    return {
+      minDigits: normalizedMinDigits,
+      maxDigits: normalizedMaxDigits
+    }
+  }
+
+  normalizeDigitLimitsForPeriod (min, max, minSegmentSize) {
+    const minProvided = min !== undefined
+    const maxProvided = max !== undefined
+    function throwError () {
+      throw new Error(`conflict: minDigits (${min}) > maxDigits (${max})`)
+    }
+
+    if (max === 0) {
+      return [minProvided ? min : minSegmentSize, 0]
+    } else if (minProvided && maxProvided) {
+      return (min > max) ? throwError() : [min, max]
+    } else if (minProvided && !maxProvided) {
+      return [min, (min > minSegmentSize) ? min : minSegmentSize]
+    } else if (!minProvided && maxProvided) {
+      return [(minSegmentSize > max) ? max : minSegmentSize, max]
+    } else {
+      return [minSegmentSize, maxProvided ? max : minSegmentSize]
+    }
+  }
+
+  /**
+   * Calculates maximum value for given period.
+   * @param {string} period
+   * @return {number}
+   */
+  getPeriodMaxValue (period) {
+    switch (period) {
+      case 'seconds':
+      case 'minutes':
+        return 59
+      case 'hours':
+        return 23
+      case 'days':
+        if (period === this.props.minPeriod) {
```

Comment:
Now I'm thinking that "days" should be limited by 99. The limit should be there because that's what the user would expect (unless they set `maxDigits` to 0).